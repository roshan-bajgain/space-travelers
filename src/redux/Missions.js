// const Missions = () => {};
export const JOIN_MISSION = "space/missions/JOINMISSION";
export const LOAD_MISSION = "space/missions/LOADMISSION";

export const joinMission = (id) => ({
  type: JOIN_MISSION,
  payload: id,
});
export const loadMission = (id) => ({
  type: LOAD_MISSION,
  payload: id,
});

export const newStateToJoinMission = (missions, id) => {
  const newState = missions.map((mission) => {
    if (mission.mission_id !== id) {
      return mission;
    }
    return { ...mission, isReserved: true };
  });
  return newState;
};

const initialState = [];

const missionReducer = (state = initialState, action) => {
  switch (action.type) {
    case LOAD_MISSION:
      return [...state, ...action.payload];
    case JOIN_MISSION:
      return newStateToJoinMission(state, action.payload);

    default:
      return state;
  }
};
export default missionReducer;
