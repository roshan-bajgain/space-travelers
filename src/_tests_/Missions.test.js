import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../redux/configureStore";
import Missions from "../components/Space-missions";

describe("Mission Element Tests", () => {
  test("Check if Component Renders in DOM", () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>
    );
    expect(screen.queryByText(/Rocket/)).toBeNull();
    expect(screen.queryByText(/Mission/)).toBeTruthy();
  });
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
