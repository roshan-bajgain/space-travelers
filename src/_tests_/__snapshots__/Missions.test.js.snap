// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Element Tests renders correctly 1`] = `
<div>
  <table
    className="table table table-sm table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
           Mission 
        </th>
         
        <th>
           Description 
        </th>
         
        <th>
           Status 
        </th>
         
        <th>
           
           
        </th>
         
      </tr>
       
    </thead>
     
    <tbody>
       
       
    </tbody>
     
  </table>
   
</div>
`;
